pkg/ddl/partition.go

```diff
@@ -276,8 +276,13 @@ func alterTablePartitionBundles(t *meta.Mutator, tblInfo *model.TableInfo, addin
 	// tblInfo do not include added partitions, so we should add them first
 	tblInfo = tblInfo.Clone()
 	p := *tblInfo.Partition
-	p.Definitions = append([]model.PartitionDefinition{}, p.Definitions...)
-	p.Definitions = append(tblInfo.Partition.Definitions, addingDefinitions...)
+	if p.DDLAction == model.ActionAlterTablePartitioning && p.Type == pmodel.PartitionTypeNone {
+		// skip adding the original table as partition
+		p.Definitions = []model.PartitionDefinition{}
+	} else {
+		p.Definitions = append([]model.PartitionDefinition{}, p.Definitions...)
```

Comment:
Seems this branch is useless.

Reply:
It is the same as the original code, most likely only to replace the original slice struct. I will check if I can remove it.